File: Documentation/6.0/Samples/nodejs/client-api/commands/documents/delete.js
import { DocumentStore } from "ravendb";

const documentStore = new DocumentStore();

async function deleteDocumentsCommand() {
    {
        //region delete_document_1
        // Define the Delete Command
        // Pass the document ID & whether to make a concurrency check
        const command = new DeleteDocumentCommand("employees/1-A", null);

        // Send the command to the server using the Store's Request Executor
        await documentStore.getRequestExecutor().execute(command);
        //endregion
    }
    {
        //region delete_document_2
        const command = new DeleteDocumentCommand("employees/1-A", null);

        // Send the command to the server using the Session's Request Executor
        await session.advanced.requestExecutor.execute(command);
        //endregion
    }
    {
        //region delete_document_3
        // Load a document
        const employeeDocument = await session.load('employees/2-A');
        const cv = session.advanced.getChangeVectorFor(employeeDocument);
        
        // Modify the document content and save changes
        // The change-vector of the stored document will change
        employeeDocument.Title = "Some new title";
        await session.saveChanges();

        try  {
            // Try to delete the document with the previous change-vector
            const command = new DeleteDocumentCommand("employees/2-A", cv);
            await session.advanced.requestExecutor.execute(command);
        }
        catch (err) {
            // A concurrency exception is thrown 
            // since the change-vector of the document in the database
            // does not match the change-vector specified in the delete command
            assert.equal(err.name, "ConcurrencyException");
        }
        //endregion
    }
}

{
    //region syntax
    DeleteDocumentCommand(id, changeVector);
    //endregion
}

File: Documentation/6.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Commands/Documents/Delete.cs
using System;
using System.Threading.Tasks;
using Raven.Client.Documents;
using Raven.Client.Documents.Commands;
using Raven.Documentation.Samples.Orders;
using Xunit;

namespace Raven.Documentation.Samples.ClientApi.Commands.Documents
{
    public class DeleteSamples
    {
        public async Task ExamplesWithStore()
        {
            #region delete_document_1
            using (var store = new DocumentStore())
            using (store.GetRequestExecutor().ContextPool.AllocateOperationContext(out var context))
            {
                var command = new DeleteDocumentCommand("employees/1-A", null);
                store.GetRequestExecutor().Execute(command, context);
            }
            #endregion
            
            #region delete_document_1_async
            using (var store = new DocumentStore())
            using (store.GetRequestExecutor().ContextPool.AllocateOperationContext(out var context))
            {
                var command = new DeleteDocumentCommand("employees/1-A", null);
                await store.GetRequestExecutor().ExecuteAsync(command, context);
            }
            #endregion
        }

        public async Task ExamplesWithSession()
        {
            using (var store = new DocumentStore())
            {
                using (var session = store.OpenSession())
                {
                    #region delete_document_2
                    var command = new DeleteDocumentCommand("employees/1-A", null);
                    session.Advanced.RequestExecutor.Execute(command, session.Advanced.Context);
                    #endregion
                }

                using (var asyncSession = store.OpenAsyncSession())
                {
                    #region delete_document_2_async
                    var command = new DeleteDocumentCommand("employees/1-A", null);
                    await asyncSession.Advanced.RequestExecutor.ExecuteAsync(command, asyncSession.Advanced.Context);
                    #endregion
                }
                
                using (var session = store.OpenSession())
                {
                    #region delete_document_3
                    // Load a document
                    var employeeDocument = session.Load<Employee>("employees/2-A");
                    var cv = session.Advanced.GetChangeVectorFor(employeeDocument);

                    // Modify the document content and save changes
                    // The change-vector of the stored document will change
                    employeeDocument.Title = "Some new title";
                    session.SaveChanges();
                    
                    try
                    {
                        // Try to delete the document with the previous change-vector
                        var command = new DeleteDocumentCommand("employees/2-A", cv);
                        session.Advanced.RequestExecutor.Execute(command, session.Advanced.Context);
                    }
                    catch (Exception e)
                    {
                        // A concurrency exception is thrown
                        // since the change-vector of the document in the database
                        // does not match the change-vector specified in the delete command
                        Assert.IsType<Raven.Client.Exceptions.ConcurrencyException>(e);
                    }
                    #endregion
                }
                
                using (var asyncSession = store.OpenAsyncSession())
                {
                    #region delete_document_3_async
                    // Load a document
                    var employeeDocument = await asyncSession.LoadAsync<Employee>("employees/2-A");
                    var cv = asyncSession.Advanced.GetChangeVectorFor(employeeDocument);

                    // Modify the document content and save changes
                    // The change-vector of the stored document will change
                    employeeDocument.Title = "Some new title";
                    asyncSession.SaveChangesAsync();
                    
                    try
                    {
                        // Try to delete the document with the previous change-vector
                        var command = new DeleteDocumentCommand("employees/2-A", cv);
                        await asyncSession.Advanced.RequestExecutor.ExecuteAsync(command, asyncSession.Advanced.Context);
                    }
                    catch (Exception e)
                    {
                        // A concurrency exception is thrown
                        // since the change-vector of the document in the database
                        // does not match the change-vector specified in the delete command
                        Assert.IsType<Raven.Client.Exceptions.ConcurrencyException>(e);
                    }
                    #endregion
                }
            }
        }
    }
    
    public class DeleteInterfaces
    {
        private class DeleteDocumentCommand
        {
            #region syntax
            public DeleteDocumentCommand(string id, string changeVector)
            #endregion
            {
            }
        }
    }
}
